flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOperationUtils.java

```diff
@@ -230,6 +252,19 @@ private static ColumnFamilyHandle createColumnFamily(
         }
     }
 
+    private static ColumnFamilyHandle createColumnFamilyWithImport(
+            ColumnFamilyDescriptor columnDescriptor,
+            RocksDB db,
+            List<ExportImportFilesMetaData> metaDataList) {
+        try {
+            return db.createColumnFamilyWithImport(
+                    columnDescriptor, new ImportColumnFamilyOptions(), metaDataList);
+        } catch (RocksDBException e) {
+            IOUtils.closeQuietly(columnDescriptor.getOptions());
+            throw new FlinkRuntimeException("Error creating ColumnFamilyHandle.", e);
```

Comment:
Missing `metaDataList.forEach(IOUtils::closeQuietly);` ?

Reply:
I modified the code in a different way that makes sure the cleanup happens through the owner of the objects.